experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -257,3 +257,158 @@ function isomorphism(::Type{PermGroup}, W::WeylGroup; set_properties::Bool=true)
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:B, 3)
+Weyl group
+  of root system of rank 3
+    of type B3
+
+julia> P1, f1 = parabolic_subgroup(W, [1, 2])
+(Weyl group of root system of type A2, Map: P1 -> W)
+
+julia> f1(P1[1] * P1[2]) == W[1] * W[2]
+true
+
+julia> P2, f2 = parabolic_subgroup(W, [1, 2], W[1])
+(Weyl group of root system of type A2, Map: P2 -> W)
+
+julia> f2(P2[1]) == W[1] && f2(P2[2]) == W[1] * W[2] * W[1]
+true
+
+julia> P3, f3 = parabolic_subgroup(W, [1,3,2])
+(Weyl group of root system of type B3 (non-canonical ordering), Map: P3 -> W)
+
+julia> f3(P3[2]) == W[3]
+true
+```
+"""
+function parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem=one(W))
+  @req allunique(vec) "Elements of vector are not pairwise distinct"
+  @req all(i -> 1 <= i <= number_of_generators(W), vec) "Invalid indices"
+  cm = cartan_matrix(root_system(W))[vec, vec]
+  para = weyl_group(cm)
+  genimgs = map(x -> inv(w) * x * w, gens(W)[vec])
+  emb = function (u::WeylGroupElem)
+    return map_word(u, genimgs)
+  end
+  return para, MapFromFunc(para, W, emb)
+end
+
+@doc raw"""
+    parabolic_subgroup_with_projection(W::WeylGroup, vec::Vector{<:Integer}) -> WeylGroup, Map{WeylGroup, WeylGroup}, Map{WeylGroup, WeylGroup}
+
+Return a triple `(P, emb, proj)` that describes a factor of `W`, that is,
+a product of irreducible factors.
+Here `P, emb = `[`parabolic_subgroup`](@ref)`(W, vec)`
+and `proj` is the projection map from `W` onto `P`,
+which is a left-inverse of `emb`.
+If `P` is not a factor of `W`, an error occurs.
+
+# Examples
+```jldoctest
+julia> W = weyl_group([(:A, 3), (:B, 3)])
+Weyl group
+  of root system of rank 6
+    of type A3 x B3
+
+julia> P1, f1, p1 = parabolic_subgroup_with_projection(W, [1,2,3])
+(Weyl group of root system of type A3, Map: P1 -> W, Map: W -> P1)
+
+julia> p1(W[1]*W[4]*W[2]*W[6]) == P1[1] * P1[2]
+true
+
+julia> P2, f2, p2 = parabolic_subgroup_with_projection(W, [4,6,5])
+(Weyl group of root system of type B3 (non-canonical ordering), Map: P2 -> W, Map: W -> P2)
+
+julia> p2(W[5]) == P2[3]
+true
+```
+"""
+function parabolic_subgroup_with_projection(
+  W::WeylGroup, vec::Vector{<:Integer}; check_factor::Bool=true
+)
+  if check_factor
+    # Check that every generator in gens(W)[vec] commutes with every other generator.
+    # In other words, vec describes a union of irreducible components of the Coxeter diagram.
+    cm = cartan_matrix(root_system(W))
+    for i in vec
+      for j in setdiff(1:number_of_generators(W), vec)
+        @req is_zero_entry(cm, i, j) begin
+          "Input vector does not describe a factor of the Weyl group, so there is no projection homomorphism"
+        end
+      end
+    end
+  end
+
+  factor, emb = parabolic_subgroup(W, vec)
+  # Generators of W are mapped to the corresponding generators of factor,
+  # or to 1 if there is no corresponding generator
+  proj_gen_imgs = [one(factor) for _ in 1:number_of_generators(W)]
```

Comment:
```suggestion
  proj_gen_imgs = fill(one(factor), number_of_generators(W))
```
or (my personal preference but it's a matter of taste)
```suggestion
  proj_gen_imgs = fill(one(factor), ngens(W))
```

Reply:
Note that there is a semantical difference here. `fill` uses the same object for all positions, while the list comprehension uses different (but identical) objects.

So the former is slightly more performant, but may not be used in contexts with mutating functions. But as far as I can see that is not the case here

Reply:
Ah, good. I had wondered whether there was a better way to initialise the vector, thanks.